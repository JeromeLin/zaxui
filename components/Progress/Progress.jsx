import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Spinner from '../Spinner';

class Progress extends PureComponent {

  render() {
    const { prefixCls, className, theme, type, strokeWidth, percent, children, ...others } = this.props;

    const cls = classnames({
      [`${prefixCls}`]: true,
      [`theme-${theme}`]: !!theme,
      [`${prefixCls}-${type}`]: !!type,
      [className]: !!className,
    });

    const innerRender = type === 'circle'
      ? <div className={`${prefixCls}-inner`}><Spinner theme={theme} strokeWidth={strokeWidth} percent={percent} /></div>
      : <div className={`${prefixCls}-inner`} style={{ height: strokeWidth }}><div className={`${prefixCls}-bg`} style={{ width: `${percent}%` }} /></div>
      ;

    return (
      <div className={cls} {...others}>
        {innerRender}
        {children ? <div className={`${prefixCls}-text`}>{children}</div> : null}
      </div>
    );
  }
}

Progress.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  theme: Spinner.propTypes.theme,
  type: PropTypes.oneOf(['line', 'circle']),
  strokeWidth: Spinner.propTypes.strokeWidth,
  percent: Spinner.propTypes.percent,

};

Progress.defaultProps = {
  prefixCls: 'za-progress',
  className: null,
  theme: Spinner.defaultProps.theme,
  type: 'line',
  strokeWidth: Spinner.defaultProps.strokeWidth,
  percent: Spinner.defaultProps.percent,
};

export default Progress;
