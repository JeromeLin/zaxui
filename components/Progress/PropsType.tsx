interface IBase {
  theme?: 'default' | 'primary' | 'info' | 'success' | 'warning' | 'error';
  strokeWidth?: number;
  percent?: number;
  shape?: 'line' | 'circle';
}

export interface IProgress extends IBase {
  prefixCls?: string;
  className?: string;
}
