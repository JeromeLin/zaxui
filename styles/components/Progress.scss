$prefixCls: za-progress;

.#{$prefixCls} {
  display: flex;
  align-items: center;
  position: relative;

  &-inner {
    flex: 1;
    background-color: #e6e6e6;
  }

  &-text {
    display: inline-block;
    margin-left: r(5);
    font-size: r(12);
  }

  &-bg {
    height: 100%;
    background-color: $theme-primary;
    transition: width .3s;
  }

  &-circle {
    display: inline-block;
    width: r(150);
    height: r(150);

    .#{$prefixCls} {
      &-inner {
        background-color: transparent;

        svg {
          width: 100%;
          height: 100%;
        }
      }

      &-text {
        display: block;
        position: absolute;
        top: 50%;
        left: 0;
        width: 100%;
        margin: 0;
        text-align: center;
        line-height: 1;
        transform: translateY(-50%);
        font-size: r(30);
      }
    }
  }
}
