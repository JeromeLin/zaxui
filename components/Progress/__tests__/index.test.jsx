import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Progress from '../index';

describe('Progress', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Progress percent={10}>foo</Progress>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
