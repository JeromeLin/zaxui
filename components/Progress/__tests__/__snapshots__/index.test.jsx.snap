// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders correctly 1`] = `
<div
  class="za-progress theme-primary shape-line"
>
  <div
    class="za-progress-inner"
    style="height:4px;"
  >
    <div
      class="za-progress-bg"
      style="width:10%;"
    />
  </div>
</div>
`;

exports[`Progress renders has children correctly 1`] = `
<div
  class="za-progress theme-primary shape-line"
>
  <div
    class="za-progress-inner"
    style="height:4px;"
  >
    <div
      class="za-progress-bg"
      style="width:10%;"
    />
  </div>
  <div
    class="za-progress-text"
  >
    foo
  </div>
</div>
`;

exports[`Progress renders shape is circle correctly 1`] = `
<div
  class="za-progress theme-primary shape-circle"
>
  <div
    class="za-progress-inner"
  >
    <svg
      class="za-spinner theme-primary"
      viewbox="0 0 100 100"
    >
      <circle
        class="za-spinner-path"
        cx="50"
        cy="50"
        fill="none"
        r="48"
        style="stroke-width:4;"
      />
      <circle
        class="za-spinner-line"
        cx="50"
        cy="50"
        fill="none"
        r="48"
        style="stroke-dasharray:30.15928947446201 301.59289474462014;stroke-width:4;"
      />
    </svg>
  </div>
  <div
    class="za-progress-text"
  >
    foo
  </div>
</div>
`;
